src/DataAccessLayer/Repositories/BookingDbRepository.cs

```diff
@@ -0,0 +1,65 @@
+﻿using DataAccessLayer.DTO;
+using DataAccessLayer.Interfaces;
+using DataAccessLayer.Models;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer.Repositories
+{
+    public class BookingDbRepository : BaseDbRepository, IBookingRepository
+    {
+        public BookingDbRepository(EfCoreContext efContext) : base(efContext)
+        {
+        }
+
+        public async Task<BookingDto> Add(BookingDto booking, CancellationToken cancellationToken = default)
+        {
+            //booking.Id = Guid.NewGuid();
+            var bookingEntity = await _dbContext.Bookings.AddAsync(booking, cancellationToken);
+            await _dbContext.SaveChangesAsync(cancellationToken);
+            return bookingEntity.Entity;
+        }
+
+        public async Task<(IQueryable<BookingDto> FilteredItems, int TotalCount)> GetAll(ItemsRequest request, CancellationToken cancellationToken = default)
+        {
+            var query = _dbContext.Bookings.Include(e => e.Products).AsNoTracking();
+            if (!string.IsNullOrEmpty(request.ItemName))
+            {
+                query = query.Where(item => item.Name.Contains(request.ItemName));
+            }
+            // bottleneck ??
+            int totalCount = await query.CountAsync(cancellationToken);
```

Comment:
EntityFramework should translate the CountAsync method into a native query like SELECT COUNT(*) FROM ... which has no problems with the performance even on the huge datasets